### PR TITLE
website-gatsby: An ocular-gatsby based website

### PR DESCRIPTION
**Pull request by ibgreen**
This is an initial setup of an ocular-gatsby based website. The docs and TOC are generated but they are not processed correctly due to the way they are generated by `documentation.js`.

The setup was copy-pasted from loaders.gl so there may be some copy-pasta left.

### CODE REVIEW

**Comment by ibgreen** [comment]
@macrigiuseppe @heshan0131 to continue with this we will likely need to agree on some changes to the current doc structure. 

I don't want to go ahead without pre-approval. If there is interest I can start summarizing the issues/decision point.

To test, just run `yarn` and `yarn start` in `website-gatsby`

File: website-gatsby/templates/index.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  window.website = true;
+}
+
+const HeroExample = 'div';
+
+// require('./example-gltf').default;
+
+export default class IndexPage extends React.Component {
+  render() {
+    return <Home HeroExample={HeroExample} />;
```
Comment (heshan0131) [inline website-gatsby/templates/index.jsx]: kepler.gl home page has plenty of js react component that does fancy fade in and fade out base don scroll. Does this mean we can still keep our current `Home` [component](https://github.com/keplergl/kepler.gl/blob/master/website/src/components/home.js)?

**Comment by heshan0131** [comment]
I guess the biggest question here is how to load the demo app at kepler.gl/demo. Currently, the website uses react route to mount demo-app at `/demo`. How are we going to keep this behavior when using gatsby?

File: website-gatsby/.eslintignore [hunk 2]
```diff
@@ -0,0 +1 @@
+../modules
```
Comment (macrigiuseppe) [inline website-gatsby/.eslintignore]: add new line

File: website-gatsby/templates/index.jsx (hunk 1 above)
Comment (macrigiuseppe) [reply website-gatsby/templates/index.jsx]: +1

Comment (ibgreen) [reply website-gatsby/templates/index.jsx]: >  Does this mean we can still keep our current Home component?

Yes of course. Gatsby is 100% React based and is designed to let you wire up any React rendered components and render them as separate static HTML+JS pages. 

I have now wired up gatsby with redux and initial webpack settings but more work is needed. However it turns out that kepler has **a lot** of special build settings etc, it was too much for me to handle in just 2 hours.

@macrigiuseppe @heshan0131  can we collaborate on finishing up the home page component integration?

Just go to `website-gatsby`, uncomment the import in `templates/index.js`  do `yarn` and `yarn start` and start fixing errors :)

If we do it together I think we can get it done quickly...

**Comment by ibgreen** [comment]
> I guess the biggest question here is how to load the demo app at kepler.gl/demo. Currently, the website uses react route to mount demo-app at `/demo`. How are we going to keep this behavior when using gatsby? 

React router is no longer necessary, as each Gatsby page generates its own HTML. So it should be possible to add an href to demo, alternatively Gatsby also has a <Link> component.

Or we may be able to just build the demo as part of the website, although that will make the website build bigger and slower...

**Comment by ibgreen** [comment]
I'd like to land this (it is in a separate folder) and iterate on it together.

I recommend you both look at the docs generated by this setup. In particular the many (otherwise excellent) screen shot images are now too big and load quite slowly. We can resize them manually or use a gatsby-image-sharp setup to preprocess them.

**Pull request merged**